### ISSUE TITLE
update extensions from private repo

### ISSUE THREAD

**Issue by sanzoghenzo** [eirannejad/pyRevit#964]
**Describe the bug**

Following #801, I was able to deploy my extension from a private git repo (BTW, I think that you should note somewhere that the root of the repo should be the ".extension" folder, I assumed that the folder should exist inside it).

If I try to update the extension via CLI, I get `Error: request failed with status code: 401`.
`--username` and `--password` are not available for `pyrevit extensions` subcommands.

**To Reproduce**

1. install the extension from a private repo (e.g. `pyrevit extend ui myextension https://test.gitlab.domain.com/group/myextension.git --branch=test_branch --username=user --password=verystrong`)
2. check that the extension is correctly listed with `pyrevit extensions`
3. try to update it with `pyrevit update myextension`
4. Get `Error: request failed with status code: 401`.

**Expected behavior**
The extension manager should remember the username and password or provide the ability to specify them.

**Desktop (please complete the following information):**
 - OS: windows 10 64 bit 1909
 - pyRevit Version 4.8.0 beta
 - pyRevit Environment: 
```
==> Registered Clones (full git repos)
==> Registered Clones (deployed from archive/image)
master | Deploy: "basepublic" | Branch: "master" | Version: "4.8.0" | Path: "C:\Users\a.ghensi\AppData\Roaming\pyRevit-Master"
==> Attachments
master | Product: "2020.2 (Update)" | Engine: 277 | Path: "C:\Users\a.ghensi\AppData\Roaming\pyRevit-Master" | Manifest: "C:\Users\a.ghensi\AppData\Roaming\Autodesk\Revit\Addins\2020\pyRevit.addin"
master | Product: "2019 First Customer Ship" | Engine: 277 | Path: "C:\Users\a.ghensi\AppData\Roaming\pyRevit-Master" | Manifest: "C:\Users\a.ghensi\AppData\Roaming\Autodesk\Revit\Addins\2019\pyRevit.addin"
==> Installed Extensions
pyRevitSWS | Type: Unknown | Repo: "https://git.sws-digital.com/bim/pyrevitsws.git" | Installed: "C:\Users\a.ghensi\AppData\Roaming\pyRevit\Extensions\pyRevitSWS.extension"
==> Default Extension Search Path
C:\Users\a.ghensi\AppData\Roaming\pyRevit\Extensions
==> Extension Search Paths
C:\Users\a.ghensi\AppData\Roaming\pyRevit\Extensions
==> Extension Sources - Default
https://github.com/eirannejad/pyRevit/raw/master/extensions/extensions.json
==> Extension Sources - Additional
==> Installed Revits
2020.2 (Update) | Version: 20.2.0.48 | Build: 20191031_1115(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2020"
2019 First Customer Ship | Version: 19.0.0.405 | Build: 20180216_1515(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2019"
==> Running Revit Instances
PID: 27188 | 2019 First Customer Ship | Version: 19.0.0.405 | Build: 20180216_1515(x64) | Language: 0 | Path: "C:\Program Files\Autodesk\Revit 2019"
==> User Environment
Microsoft Windows 10 [Version 10.0.18363]
Executing User: SWSGLOBAL\a.ghensi
Active User: SWSGLOBAL\a.ghensi
Admin Access: No
%APPDATA%: "C:\Users\a.ghensi\AppData\Roaming"
Latest Installed .Net Framework: 4.8
Installed .Net Target Packs: v3.5 v4.0 v4.5 v4.5.1 v4.5.2 v4.6 v4.6.1 v4.6.2 v4.7 v4.7.2 v4.X
Installed .Net-Core Target Packs: v1.0.2 v2.1.600-preview-009497 v2.1.700-preview-009618 v2.1.800-preview-009696 v3.1.201
pyRevit CLI 4.8.0.0
```

**Additional context**

The goal here is to ease the deployment and update process for my colleagues;

It would be wonderful if the update process can be done directly in the GUI (maybe with the pyrevit update button).

**Comment by jmcouffin** [comment]
**Closing**
Issue older than a year.
Thanks for pointing it out.
If you feel it is still relevant, feel free to reopen the issue

**Comment by sanzoghenzo** [comment]
hi @jmcouffin,
I don't understand why these old issues are being closed; They are still relevant.
For this specific issue, as a workaround I had to instruct my colleagues on how to install and use git just to pull updates :sweat: 
I don't know if something has changed in the source code since this issue was opened, and I didn't test how it works with personal access tokens (we enabled 2FA in our gitlab instance, so one should use a PAT in a git client).
I'll make a few tests and let you know.
